packages/teams-js/test/public/navigation.spec.ts

```diff
@@ -21,134 +22,173 @@ describe('MicrosoftTeams-Navigation', () => {
 
-    it('should allow calls from task context', async () => {
-      await utils.initializeWithContext('task');
-
-      navigateCrossDomain('https://valid.origin.com');
+    Object.values(FrameContexts).forEach(context => {
+      it(`navigation.navigateToTab should register the navigateToTab action when initialized with ${context} context`, () => {
+        utils.initializeWithContext(context);
+        navigateToTab(null);
+        const navigateToTabMsg = utils.findMessageByFunc('navigateToTab');
+        expect(navigateToTabMsg).not.toBeNull();
+      });
     });
+  });
 
-    it('should allow calls from stage context', async () => {
-      await utils.initializeWithContext('stage');
-
-      navigateCrossDomain('https://valid.origin.com');
+  describe('Testing navigation.navigateCrossDomain function', () => {
+    const allowedContexts = [
+      FrameContexts.content,
+      FrameContexts.sidePanel,
+      FrameContexts.settings,
+      FrameContexts.remove,
+      FrameContexts.task,
+      FrameContexts.stage,
+      FrameContexts.meetingStage,
+    ];
+
+    it('navigation.navigateCrossDomain should not allow calls before initialization', () => {
+      expect(() => navigateCrossDomain('https://valid.origin.com')).toThrowError(
+        'The library has not yet been initialized',
+      );
     });
 
-    it('should successfully navigate cross-origin', async () => {
-      await utils.initializeWithContext('content');
+    Object.values(FrameContexts).forEach(context => {
+      if (allowedContexts.some(allowedContexts => allowedContexts === context)) {
+        it(`navigation.navigateCrossDomain should allow calls when initialized with ${context} context`, async () => {
+          await utils.initializeWithContext(context);
 
-      navigateCrossDomain('https://valid.origin.com');
+          navigateCrossDomain('https://valid.origin.com');
+        });
 
-      const navigateCrossDomainMessage = utils.findMessageByFunc('navigateCrossDomain');
-      expect(navigateCrossDomainMessage).not.toBeNull();
-      expect(navigateCrossDomainMessage.args.length).toBe(1);
-      expect(navigateCrossDomainMessage.args[0]).toBe('https://valid.origin.com');
-    });
+        it(`navigation.navigateCrossDomain should successfully navigate cross-origin when initialized with ${context} context`, async () => {
+          await utils.initializeWithContext(context);
 
-    it('should throw on invalid cross-origin navigation request', done => {
-      utils.initializeWithContext('settings').then(() => {
-        navigateCrossDomain('https://invalid.origin.com', (success, reason) => {
-          expect(success).toBeFalsy();
-          expect(reason).toBe(
-            'Cross-origin navigation is only supported for URLs matching the pattern registered in the manifest.',
-          );
-          done();
+          navigateCrossDomain('https://valid.origin.com');
+
+          const navigateCrossDomainMessage = utils.findMessageByFunc('navigateCrossDomain');
```

Comment:
same here and other APIs in this file. I am sorry if I am getting it totally wrong. I think other people's review might help!